projects/storefrontlib/src/cms-components/checkout/services/checkout-step.service.ts

```diff
@@ -48,47 +48,110 @@ export class CheckoutStepService {
   }
 
   back(activatedRoute: ActivatedRoute): void {
-    const previousUrl = this.checkoutConfigService.getPreviousCheckoutStepUrl(
-      activatedRoute
-    );
+    const previousUrl = this.getPreviousCheckoutStepUrl(activatedRoute);
     this.routingService.go(previousUrl === null ? 'cart' : previousUrl);
   }
 
   next(activatedRoute: ActivatedRoute): void {
-    const nextUrl = this.checkoutConfigService.getNextCheckoutStepUrl(
-      activatedRoute
-    );
+    const nextUrl = this.getNextCheckoutStepUrl(activatedRoute);
     this.routingService.go(nextUrl);
   }
 
   getBackBntText(activatedRoute: ActivatedRoute): string {
-    if (
-      this.checkoutConfigService.getPreviousCheckoutStepUrl(activatedRoute) ===
-      null
-    ) {
+    if (this.getPreviousCheckoutStepUrl(activatedRoute) === null) {
       return 'checkout.backToCart';
     }
-
     return 'common.back';
   }
 
   resetSteps(): void {
-    this.intialSteps = this.checkoutConfigService.steps
+    this.allSteps = this.checkoutConfigService.steps
       .filter((step) => !step.disabled)
       .map((x) => Object.assign({}, x));
-    this.steps$.next(this.intialSteps);
+    this.steps$.next(this.allSteps);
   }
 
   disableEnableStep(
     currentStepType: CheckoutStepType,
     disabled: boolean
   ): void {
-    const currentStep = this.intialSteps.find((step) =>
+    const currentStep = this.allSteps.find((step) =>
       step.type.includes(currentStepType)
     );
     if (currentStep) {
       currentStep.disabled = disabled;
-      this.steps$.next(this.intialSteps.filter((step) => !step.disabled));
+      this.steps$.next(this.allSteps.filter((step) => !step.disabled));
+    }
+  }
+
+  getNextCheckoutStepUrl(activatedRoute: ActivatedRoute): string {
+    const stepIndex = this.getCurrentStepIndex(activatedRoute);
```

Comment:
Do we handle the case where `stepIndex` is null?

Reply:
if `stepIndex` is null, `if (stepIndex >= 0)` is false.

Reply:
will add it. thanks.